src/Domain/Structure/Direction.hpp

```diff
@@ -25,28 +25,35 @@ class Direction {
   static constexpr const size_t volume_dim = VolumeDim;
 
   /// The logical-coordinate names of each dimension
-  enum class Axis;
+  enum class Axis : uint8_t;
 
   /// Construct by specifying an Axis and a Side.
-  Direction(Axis axis, Side side) : axis_(axis), side_(side) {}
+  Direction(Axis axis, Side side);
 
   /// Construct by specifying a dimension and a Side.
   Direction(size_t dimension, Side side);
 
   /// Default constructor for Charm++ serialization.
   Direction();
 
+  /// Get a Direction representing "self" or "no direction"
+  static Direction<VolumeDim> self();
+
+  /// Get the bits representing a Direction that represents "self" or "no
+  /// direction"
+  static uint8_t self_bits();
+
   /// The dimension of the Direction
-  size_t dimension() const { return static_cast<size_t>(axis_); }
+  size_t dimension() const { return static_cast<uint8_t>(axis()); }
 
   /// The Axis of the Direction
-  Axis axis() const { return axis_; }
+  Axis axis() const { return static_cast<Axis>(bit_field_ bitand 0b0011); }
 
   /// The side of the Direction
-  Side side() const { return side_; }
+  Side side() const { return static_cast<Side>(bit_field_ bitand 0b1100); }
 
   /// The sign for the normal to the Side.
-  double sign() const { return (Side::Lower == side_ ? -1.0 : 1.0); }
+  double sign() const { return (Side::Lower == side() ? -1.0 : 1.0); }
```

Comment:
this will give 1.0 for Side::Self and Side::Uninitialized, should we add an ASSERT?

Reply:
Yea, good idea. I'll add an `ASSERT`